### PR TITLE
Create DecisionReviewUpdated::UpdatedIssueCollectionBuilder

### PR DESCRIPTION
**Pull request by gcd253**
Resolves [APPEALS-55441](https://jira.devops.va.gov/browse/APPEALS-55441)

# Description
- Creates a class `DecisionReviewUpdated::UpdatedIssueCollectionBuilder` which identifies updated issues that have a `reasonForContentionAction` with a value of `"PRIOR_DECISION_TEXT_CHANGED"` and a `contentionAction` with a value of `"NONE"` or a value of `"UPDATE_CONTENTION"`. Issues with any other values for these attributes will be ignored.

- Includes new scenarios in the Kafka Message Generator. All added scenarios address the following attributes:
  - `reasonForContentionAction` has a value of `"PRIOR_DECISION_TEXT_CHANGED"` and a `contentionAction` has a value of `"NONE"`

- Adds `edited_description` field to `DecisionReviewUpdated::RequestIssue` and `DecisionReviewUpdated::RequestIssueBuilder`
  - Attribute `edited_description` is only assigned a value when `reasonForContentionAction` has a value of `"PRIOR_DECISION_TEXT_CHANGED"` and a `contentionAction` has a value of `"UPDATE_CONTENTION"`

## Acceptance Criteria
- [x] `build_issues` method exists and is defined:
  - [x] Method needs to identify `DecisionReviewIssueUpdated` issues that meet the following criteria using scopes:
    - [x] A scope titled `updated_issues` exists & has a `reasonForContentionAction` with a value of `"PRIOR_DECISION_TEXT_CHANGED"` & a `contentionAction` with a value of `"UPDATE_CONTENTION"`.  This scope only retrieves issues that are part of the `decisionReviewIssuesUpdated` array in the `DecisionReviewUpdated` event.
    - [x] A scope titled `updated_issues` exists & has a `reasonForContentionAction` with a value of `"PRIOR_DECISION_TEXT_CHANGED"` & a `contentionAction` with a value of `"NONE"`.  This scope only retrieves issues that are part of the `decisionReviewIssuesUpdated` array in the `DecisionReviewUpdated` event.
- [x] `build_request_issues` method exists and is defined:
  - [x] Method needs to have a begin/rescue block with error handling
  - [x] Method needs to call on `Builders::DecisionReviewUpdated::RequestIssueBuilder` class to generate request issues.
- [x] Update Kafka Message Generator to have a scenario where `reasonForContentionAction` has a value of `"PRIOR_DECISION_TEXT_CHANGED"` & a `contentionAction` has a value of `"NONE"` within the `decisionReviewIssuesUpdated` array in the `DecisionReviewUpdated` event.

## Testing Plan

1. [APPEALS-57455](https://jira.devops.va.gov/browse/APPEALS-57455)
2. [APPEALS-57131](https://jira.devops.va.gov/browse/APPEALS-57131)
3. [Jira Xray](https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-57455&testIssueKey=APPEALS-57131)

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: app/models/builders/decision_review_updated/updated_issue_collection_builder.rb [hunk 1]
```diff
@@ -1,5 +1,42 @@
...
+    updated_issues.map.with_index do |issue, index|
+      build_request_issue(issue, index)
+    end
+  end
+
+  def build_request_issue(issue, index)
+    begin
+      Builders::DecisionReviewUpdated::RequestIssueBuilder.build(issue, @decision_review_model, @bis_rating_profiles)
+    rescue StandardError => error
+      message = "Failed building from #{self.class} for DecisionReview Claim ID: #{@decision_review_model.claim_id} " \
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_updated/updated_issue_collection_builder.rb:17]: Please update this from 'Failed building from' to 'Failed building added_issues from'

File: app/models/builders/base_request_issue_collection_builder.rb [hunk 2]
```diff
@@ -8,11 +8,14 @@ class Builders::BaseRequestIssueCollectionBuilder
   RATING = "RATING"
 
   REASON_FOR_CONTENTION_ACTIONS = {
-    ISSUE_REMOVED: "REMOVED_SELECTED"
+    ISSUE_REMOVED: "REMOVED_SELECTED",
+    PRIOR_DECISION_TEXT_CHANGED: "PRIOR_DECISION_TEXT_CHANGED"
```
Comment (Aaron-Willis) [inline app/models/builders/base_request_issue_collection_builder.rb:12]: Please update from PRIOR_DECISION_TEXT_CHANGED: "PRIOR_DECISION_TEXT_CHANGED" to TEXT_CHANGED: "PRIOR_DECISION_TEXT_CHANGED"

File: app/models/builders/base_request_issue_collection_builder.rb [hunk 3]
```diff
@@ -8,11 +8,14 @@ class Builders::BaseRequestIssueCollectionBuilder
...
   }.freeze
 
   CONTENTION_ACTIONS = {
-    CONTENTION_DELETED: "DELETE_CONTENTION"
+    CONTENTION_DELETED: "DELETE_CONTENTION",
+    NONE: "NONE",
```
Comment (Aaron-Willis) [inline app/models/builders/base_request_issue_collection_builder.rb:17]: Can you please update from NONE: "NONE" to NO_CONTENTION_ACTION: "NONE"

File: app/models/builders/decision_review_updated/updated_issue_collection_builder.rb [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+    update_contention_issues + contention_none_issues
+  end
+
+  def update_contention_issues
+    @decision_review_model.decision_review_issues_updated.select do |issue|
+      issue.reason_for_contention_action == text_changed && issue.contention_action == contention_updated
+    end
+  end
+
+  def contention_none_issues
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_updated/updated_issue_collection_builder.rb:37]: Please rename to 'updated_issues_without_contentions'

File: spec/factories/decision_review_updated/decision_review_updated_request_issue.rb [hunk 5]
```diff
@@ -34,5 +34,10 @@
       closed_at { DateTime.new(2022, 2, 1) }
       closed_status { true }
     end
+
+    trait :updated_request_issue do
```
Comment (Aaron-Willis) [inline spec/factories/decision_review_updated/decision_review_updated_request_issue.rb:38]: I think that only 'edited_description' field would be changed here

File: app/models/builders/decision_review_updated/updated_issue_collection_builder.rb (hunk 1 above)
Comment (gcd253) [reply app/models/builders/decision_review_updated/updated_issue_collection_builder.rb:17]: I think you probably mean updated_issues, but will do

File: app/models/builders/base_request_issue_collection_builder.rb [hunk 6]
```diff
@@ -10,12 +10,17 @@ class Builders::BaseRequestIssueCollectionBuilder
   REASON_FOR_CONTENTION_ACTIONS = {
     INELIGIBLE_REASON_CHANGED: "INELIGIBLE_REASON_CHANGED",
     ISSUE_REMOVED: "REMOVED_SELECTED",
-    ISSUE_WITHDRAWN: "WITHDRAWN_SELECTED"
+    ISSUE_WITHDRAWN: "WITHDRAWN_SELECTED",
+    TEXT_CHANGED: "PRIOR_DECISION_TEXT_CHANGED",
+    NEW_ELIGIBLE_ISSUE: "NEW_ELIGIBLE_ISSUE",
```
Comment (Aaron-Willis) [inline app/models/builders/base_request_issue_collection_builder.rb:15]: Please update to ISSUE_ADDED to match our naming conventions

File: app/models/builders/base_request_issue_collection_builder.rb [hunk 7]
```diff
@@ -121,6 +126,18 @@ def removed
     REASON_FOR_CONTENTION_ACTIONS[:ISSUE_REMOVED]
   end
 
+  def text_changed
+    REASON_FOR_CONTENTION_ACTIONS[:TEXT_CHANGED]
+  end
+
+  def new_eligible_issue
```
Comment (Aaron-Willis) [inline app/models/builders/base_request_issue_collection_builder.rb:133]: please rename to issue_added

File: app/models/builders/decision_review_updated/added_issue_collection_builder.rb [hunk 8]
```diff
@@ -22,13 +22,13 @@ def build_request_issue(issue, index)
 
   def newly_added_eligible_issues
     @decision_review_model.decision_review_issues_created.select do |issue|
-      issue.reason_for_contention_action == "NEW_ELIGIBLE_ISSUE" && issue.contention_action == "ADD_CONTENTION"
+      issue.reason_for_contention_action == new_eligible_issue && issue.contention_action == contention_added
```
Comment (Aaron-Willis) [inline app/models/builders/decision_review_updated/added_issue_collection_builder.rb:25]: please update name from new_eligible_issue to issue_added

**Review by Aaron-Willis — APPROVED** [review]

**Pull request merged**